lib/fluent/plugin/output.rb

```diff
@@ -597,6 +598,43 @@ def terminate
         super
       end
 
+      def actual_flush_thread_count
+        return 0 unless @buffering
+        return @buffer_config.flush_thread_count unless @as_secondary
+        @primary_instance.buffer_config.flush_thread_count
+      end
+
+      # Run the passed block in the appropriate lock condition for multiple threads and workers.
+      # The lock between workers is made for every `worker_lock_name`.
+      # (For multiple workers, the lock is shared if `worker_lock_name` is the same value).
+      # For multiple threads, `worker_lock_name` is not used, and the lock is shared by all
+      # threads in the same process.
+      def acquire_lock_if_need(worker_lock_name)
```

Comment:
I feel it's not Rubyish naming (original `acquire_worker_lock` is so too though).
`synchronize_foobar` might be better but I don't have good concrete idea for `foobar` to express it.
(`worker_and_flush_thread` is too redundant.)

@kou Do you have any idea for it?

Reply:
`synchronize_path(path)`?
Or `guard_path(path)`?

Reply:
Thanks for your advice!

I have one point of concern.
Since the argument `path` is not used in the case of multiple threads, isn't `ooxx_path` misleadable?

Reply:
How about simply `synchronize(path)`?

Reply:
> Since the argument `path` is not used in the case of multiple threads, isn't `ooxx_path` misleadable?

Sorry but I can't understand the point. `synchronize_path(path) {...}` ensures `path` processable only by the current thread in the current process. `synchronize_path` does or does not something against `path` is details of `synchronize_path`. `synchronize_path` users should not care about it.

BTW, another candidate `use_path(path)`.

> How about simply `synchronize(worker_lock_path)`?

I object it. Because `Output`'s main responsibility isn't synchronization. So only `synchronize` is ambitious. For example, `Output` has two `Mutex`s. It means that there are at least two synchronization units in `Output`.

`Mutex#synchronize` is OK because `Mutex`'s main responsibility is one synchronization.

Reply:
Thanks for your detailed explanation! I see! I understand!

Although this does NOT provide exclusive processing for each `path` (but for all threads), it after all means that this ensures the `path` processable only be the current thread.
So we don't need to concern that the `path` is not used in the case of multiple threads.

https://github.com/fluent/fluentd/blob/f23e7e689c813a3040610e8f6b5958be3a369c28/lib/fluent/plugin/output.rb#L629-L636

Reply:
Fixed! Thanks for your advice! @ashie @kou !